### ISSUE TITLE
[CodeGen] name clashe with imported fields (EnumValue)

### ISSUE THREAD

**Issue by MasseGuillaume** [ghostdogpr/caliban#543]
Another day, another weird GraphQL schema!

I'm working with an e-commerce platform called commercetools. They have a graphql schema that looks like this:
https://github.com/commercetools/sunrise-spa/blob/master/graphql.schema.json (They are not even using this type ...).

They decided to define `EnumValue` and this clashes with the import: `import caliban.client.Value._`.

I think this is correct because the specs refer to `__EnumValue`. https://spec.graphql.org/June2018/#sec-The-__EnumValue-Type

Generated code

```scala
// evil part
type EnumValue
object EnumValue {
  def key: SelectionBuilder[EnumValue, String] = Field("key", Scalar())
  def label: SelectionBuilder[EnumValue, String] = Field("label", Scalar())
}

// usage that breaks ...
sealed trait SubscriptionHealthStatus
      extends scala.Product
      with scala.Serializable
  object SubscriptionHealthStatus {
    case object TemporaryError extends SubscriptionHealthStatus
    case object ConfigurationErrorDeliveryStopped
        extends SubscriptionHealthStatus
    case object ConfigurationError extends SubscriptionHealthStatus
    case object Healthy extends SubscriptionHealthStatus

    //...
    implicit val encoder: ArgEncoder[SubscriptionHealthStatus] =
      new ArgEncoder[SubscriptionHealthStatus] {
        override def encode(value: SubscriptionHealthStatus): Value =
          value match {
            case SubscriptionHealthStatus.TemporaryError =>
              EnumValue("TemporaryError")
            case SubscriptionHealthStatus.ConfigurationErrorDeliveryStopped =>
              EnumValue("ConfigurationErrorDeliveryStopped")
            case SubscriptionHealthStatus.ConfigurationError =>
              EnumValue("ConfigurationError")
            case SubscriptionHealthStatus.Healthy => __EnumValue("Healthy")
          }
        override def typeName: String = "SubscriptionHealthStatus"
      }
  }

```

Compilation issue

```
[error] /home/gui/automat/commercetools-graphql-scala/src/main/scala/client.scala:646:24: com.automat.product.commercetools.graphql.Client.EnumValue.type does not take parameters
[error]               EnumValue("ConfigurationErrorDeliveryStopped")
[error]                        ^
```

Proposal:

```diff
diff --git a/src/main/scala/client.scala b/src/main/scala/client.scala
index feac5e6..be290fe 100644
--- a/src/main/scala/client.scala
+++ b/src/main/scala/client.scala
@@ -5,7 +5,10 @@ import caliban.client.FieldBuilder._
 import caliban.client.SelectionBuilder._
 import caliban.client._
 import caliban.client.Operations._
-import caliban.client.Value._
+import caliban.client.Value.{
+  StringValue, ObjectValue, ListValue, NullValue, // ... and more ?
+  EnumValue => __EnumValue
+}
 
@@ -641,12 +644,12 @@ object Client {
         override def encode(value: SubscriptionHealthStatus): Value =
           value match {
             case SubscriptionHealthStatus.TemporaryError =>
-              EnumValue("TemporaryError")
+              __EnumValue("TemporaryError")
             case SubscriptionHealthStatus.ConfigurationErrorDeliveryStopped =>
-              EnumValue("ConfigurationErrorDeliveryStopped")
+              __EnumValue("ConfigurationErrorDeliveryStopped")
             case SubscriptionHealthStatus.ConfigurationError =>
-              EnumValue("ConfigurationError")
-            case SubscriptionHealthStatus.Healthy => EnumValue("Healthy")
+              __EnumValue("ConfigurationError")
+            case SubscriptionHealthStatus.Healthy => __EnumValue("Healthy")
           }
         override def typeName: String = "SubscriptionHealthStatus"
       }
```

**Comment by ghostdogpr** [comment]
How about renaming those internal classes to `__???Value` directly?

**Comment by MasseGuillaume** [comment]
Yeah, I think it's safer.

**Comment by xirc** [comment]
I'd like to work on this issue for the ZIO Hackathon :-)

**Comment by ghostdogpr** [comment]
@xirc thanks! Let me know if you have any questions.

**Comment by ghostdogpr** [comment]
Closed by #642
